### DISCUSSION TITLE
get images which are predicted correct from vals confusion mask

### DISCUSSION THREAD

**Discussion by halo34** [ultralytics/ultralytics#12686] [Q&A]
Hi,
I need the images which are right detected from yolo8.
Val provides a confusion matrix, but is it also possible to get the files which are counted as right detected in val?

**Comment by pderrenger** [comment]
@halo34 hello!

Yes, you can obtain the images that were correctly detected during validation by modifying the validation script to save these images based on their detection status. Here's a simple example of how you might adjust the validation code to save correctly detected images:

```python
from ultralytics import YOLO

# Load your model
model = YOLO('path/to/your/model.pt')

# Run validation
results = model.val()

# Accessing the correctly predicted images
correct_images = [x for x in results if x['correct']]

# Save or process these images as needed
for img in correct_images:
    img.save('path/to/save/correctly_detected_images/')
```

This code snippet assumes you have a way to determine if an image was correctly detected (`x['correct']` in this case, which you'll need to define based on your criteria for a correct detection). Adjust the paths and conditions according to your specific setup and requirements. Happy coding! 😊
